EIPS/eip-5089.md

```diff
@@ -0,0 +1,393 @@
+
+MUST revert if all of `underlyingAmount` cannot be withdrawn (due to withdrawal limit being reached, slippage, the owner not having enough principal tokens, etc).
+
+Note that some implementations will require pre-requesting to the principal token contract before a withdrawal may be performed. Those methods should be performed separately.
+
+```yaml
+- name: withdraw
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: underlyingAmount
+      type: uint256
+    - name: receiver
+      type: address
+    - name: owner
+      type: address
+
+  outputs:
+    - name: principalAmount
+      type: uint256
+```
+
+### Events
+
+#### Redeem
+
+`from` has exchanged `principalAmount` of Principal Tokens for `underlyingAmount` of underlying, and transferred that underlying to `to`.
+
+MUST be emitted when Principal Tokens are burnt and underlying is withdrawn from the contract in the `EIP5089.redeem` method.
+
+```yaml
+- name: Redeem
+  type: event
+
+  inputs:
+    - name: from
+      indexed: true
+      type: address
+    - name: to
+      indexed: true
+      type: address
+    - name: amount
+      indexed: false
+      type: uint256
+```
+
+#### Mature
+The Principal Token has hit maturity, and a a state change has been recorded on the Principal Token contract for the first time post-maturity.
+
+```yaml
+- name: Mature
+  type: event
+
+  inputs:
+    - name: timestamp
+      indexed: true
+      type: uint256
+```
+
+## Rationale
+
+The Principal Token interface is designed to be optimized for integrators with a core minimal interface alongside optional interfaces to enable backwards compatibility. Details such as accounting and management of underlying are intentionally not specified, as Principal Tokens are expected to be treated as black boxes on-chain and inspected off-chain before use.
+
+[EIP-20](./eip-20.md) is enforced as implementation details such as token approval and balance calculation directly carry over. This standardization makes Principal Tokens immediately compatible with all [EIP-20](./eip-20.md) use cases in addition to EIP-5089.
+
+All principal tokens are redeemable upon maturity, with the only variance being whether further yield is generated post-maturity. Given the ubiquity of redemption, the presence of `redeem` allows integrators to purchase Principal Tokens on an open market, and them later redeem them for a fixed-yield solely knowing the address of the Principal Token itself.
+
+This EIP draws heavily on the design of [EIP-4626](./eip-4626.md) because technically Principal Tokens could be described as a subset of Yield Bearing Vaults, extended with a `maturity` variable and restrictions on the implementation. However, extending [EIP-4626](./eip-4626.md) would force PT implementations to include methods (namely, `mint` and `deposit`) that are not necessary to the business case that PTs solve. It can also be argued that partial redemptions (implemented via `withdraw`) are rare for PTs.
```

Comment:
Another vote for removing `withdraw`!

Reply:
For simplicity's sake, i tend to agree with removing withdraw.  There are enough helper functions for an external contract to easily calculate the how many tokens should be redeemed for a given amount of underlying.

Reply:
How would you do that, without knowing the internal implementation of a given 5089, and taking into account that `convert*` functions are not necessarily exact?